x-pack/filebeat/input/http_endpoint/handler.go

```diff
@@ -59,14 +61,30 @@ func (h *httpHandler) apiResponse(w http.ResponseWriter, r *http.Request) {
 		headers = getIncludedHeaders(r, h.includeHeaders)
 	}
 
+	var responseCode int
+	var responseBody string
+
 	for _, obj := range objs {
+		var err error
+		if h.crc.provider != "" {
+			responseCode, responseBody, err = h.crc.validator(h.crc, obj)
```

Comment:
With change of `crcValidator` to the pointer type, this becomes
```suggestion
		if h.crc != nil {
			status, body, err = h.crc.validate(obj)
```